otx/algorithms/common/adapters/mmcv/hooks.py

```diff
@@ -491,25 +491,22 @@ def _init_rule(self, rule, key_indicator):
         self.key_indicator = key_indicator
         self.compare_func = self.rule_map[self.rule]
 
-    def _should_check_stopping(self, runner):
-        """Called _should_check_stopping in ReduceLROnPlateauLrUpdaterHook."""
+    def _is_check_timing(self, runner: BaseRunner) -> bool:
+        """Check whether current epoch or iter is multiple of self.interval."""
         check_time = self.every_n_epochs if self.by_epoch else self.every_n_iters
-        if not check_time(runner, self.interval):
-            # No evaluation during the interval.
-            return False
-        return True
+        return check_time(runner, self.interval)
 
     @check_input_parameters_type()
     def get_lr(self, runner: BaseRunner, base_lr: float):
         """Called get_lr in ReduceLROnPlateauLrUpdaterHook."""
-        if not self._should_check_stopping(runner) or self.warmup_iters > runner.iter:
-            return base_lr
-
         if self.current_lr < 0:
             self.current_lr = base_lr
 
-        if hasattr(runner, self.metric):
-            score = getattr(runner, self.metric, 0.0)
+        if not self._is_check_timing(runner) or self.warmup_iters > runner.iter:
```

Comment:
how about move `self.warmup_iters > runner.iter` in `self._is_check_timing()`?

Reply:
Thank you for suggestion, but if we move `self.warmup_iters > runner.iter` in `_is_check_timing()`, we need if else statement in that method. 
I believe doing one job for one function is the way to improve readability. As plan B, we can split if not~ or statement with two if statements.  ex) `if not self.is_check_timing(runner) ~ if self.warmup_iters > runner.iter`

Reply:
Reflected, thank you for your wisdom :)